torch/_dynamo/output_graph.py

```diff
@@ -547,6 +550,47 @@ def init_ambient_guards(self):
             self.guards.add(
                 GlobalStateSource().make_guard(GuardBuilder.FUNCTORCH_STACK_MATCH)
             )
+        if not torch._dynamo.compiled_autograd.in_compiled_autograd_region:
+            self.guards.add(
+                GlobalStateSource().make_guard(
+                    GuardBuilder.AUTOGRAD_SAVED_TENSORS_HOOKS
+                )
+            )
+
+    def maybe_install_saved_tensors_hooks_subgraphs(self):
+        if torch._dynamo.compiled_autograd.in_compiled_autograd_region:
+            return
+
+        hooks = torch._C._autograd._top_saved_tensors_default_hooks(True)
+        if not torch._functorch._aot_autograd.utils.top_saved_tensors_hooks_are_inlineable(
+            hooks
+        ):
+            return
+
+        p_hook, u_hook = hooks
+        inp = torch.randn(1)
+        p_gm = make_fx(p_hook)(inp)
+        p_out = p_gm(inp)
+
+        u_gm = make_fx(u_hook)(p_out)
+
+        p_g = p_gm.graph
+        u_g = u_gm.graph
+
```

Comment:
TODO(rzou): is there an alternative design that doesn't make the assumption for the shape of the inp?

Reply:
at pt2-dispatcher sync it was decided that we should tell the user to give us a graphmodule instead. This API is very explicit about how the tracing happens and unblocks the use cases. If we are able to come up with a better design after we can but it's really unclear.